### PR TITLE
Addressing API review feedback and adding unit test to ensure we can still record value higher than MaxExpectedValue

### PR DESCRIPTION
**Pull request by FabianMeiswinkel**
# Description
Addressing API review feedback and adding unit test to ensure we can still record value higher than MaxExpectedValue (which shouldn't happen but shouldn't crash app etc.)

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ClientMetricsTest.java [hunk 1]
```diff
@@ -84,6 +88,43 @@ public void afterTest() {
         this.meterRegistry = null;
     }
 
+    @Test(groups = { "unit" }, timeOut = TIMEOUT)
+    public void MaxValueExceedingDefinedLimitStillWorksWithoutException() throws Exception {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ClientMetricsTest.java:92]: nit -  name should start with lower case :)

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java [hunk 2]
```diff
@@ -30,7 +30,6 @@
 /**
  * Class with config options for Cosmos Client telemetry
  */
-@Beta(value = Beta.SinceVersion.V4_37_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public final class CosmosClientTelemetryConfig {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java:33]: nit - since beta APIs are gone now, may be remove beta import if no longer used.

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ClientMetricsTest.java (hunk 1 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/ClientMetricsTest.java]: Fixed

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java (hunk 2 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java:32]: Good point - by the way when doing this I also went through all the APIs still annotated with Beta. I feel for many of them we should simply remove the Beta tag - we won't change those APIs anymore after a year or so - and keeping the number of Beta APIs small will help in the API review negotiations.

Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosClientTelemetryConfig.java:32]: I pushed those changes to remove Beta tag - please re-review this PR

**Review by xinlian12 — APPROVED** [review]
LGTM, thanks

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java [hunk 3]
```diff
@@ -799,7 +792,6 @@ public void enableGlobalThroughputControlGroup(ThroughputControlGroupConfig grou
      *  </p>
      *
      */
-    @Beta(value = Beta.SinceVersion.V4_14_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     public void openConnectionsAndInitCaches() {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java:795]: @xinlian12 - are we ready to GA this? Or are there going to be any more behavior breaking changes?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java [hunk 4]
```diff
@@ -113,7 +113,6 @@ public Set<URI> getRegionsContacted() {
      *
      * @return set of regions contacted for this request
      */
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java:115]: Regarding the API -> 
`getRegionsContacted()` - it has been long deprecated, should we remove it?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroupConfigBuilder.java [hunk 5]
```diff
@@ -11,7 +11,6 @@
 /**
  * The throughput control group config builder.
  */
-@Beta(value = Beta.SinceVersion.V4_13_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public class ThroughputControlGroupConfigBuilder {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroupConfigBuilder.java:14]: There are multiple APIs in this class which have been deprecated for more than 5 months now. Should we also remove them ? Since we are GAing the other ones, it would be better to remove them. 
@xinlian12 thoughts?

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMetaData.java [hunk 6]
```diff
@@ -3,14 +3,12 @@
...
 import com.azure.cosmos.implementation.Utils;
-import com.azure.cosmos.util.Beta;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.core.JsonProcessingException;
 
 /**
  * Change Feed response meta data
  */
-@Beta(value = Beta.SinceVersion.V4_35_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public final class ChangeFeedMetaData {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMetaData.java:12]: Is this change intentional?
I thought we agreed on not removing Beta tags for FFCF changes. 
Same for the other model classes of FFCF. 
Since FFCF will evolve - we haven't even public previewed, I would rather not remove Beta API annotation from these models.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMode.java [hunk 7]
```diff
@@ -3,12 +3,9 @@
 
 package com.azure.cosmos.implementation.changefeed.common;
 
-import com.azure.cosmos.util.Beta;
-
 /**
  * Change feed mode
  */
-@Beta(value = Beta.SinceVersion.V4_35_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public enum ChangeFeedMode {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMode.java:9]: same here, please revert this.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedOperationType.java [hunk 8]
```diff
@@ -2,13 +2,11 @@
...
 package com.azure.cosmos.implementation.changefeed.common;
 
-import com.azure.cosmos.util.Beta;
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * Change feed operation type
  */
-@Beta(value = Beta.SinceVersion.V4_35_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public enum ChangeFeedOperationType {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedOperationType.java:10]: same here, please revert this.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorItem.java [hunk 9]
```diff
@@ -15,7 +14,6 @@
  *
  * Caller is recommended to type cast {@link JsonNode} to cosmos item structure.
  */
-@Beta(value = Beta.SinceVersion.V4_35_0, warningText = Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
 public final class ChangeFeedProcessorItem {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorItem.java:17]: same here, please revert this.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java [hunk 10]
```diff
@@ -87,8 +85,6 @@ ChangeFeedState getContinuation() {
      *
      * @return the feed range.
      */
-    @Beta(value = Beta.SinceVersion.V4_12_0, warningText =
-        Beta.PREVIEW_SUBJECT_TO_CHANGE_WARNING)
     public FeedRange getFeedRange() {
```
Comment (kushagraThapar) [inline sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java:88]: I thought that FeedRange still had some work to be done (may be in regards to merge support) before we GA it. 
@FabianMeiswinkel  @xinlian12  ?

**Review by kushagraThapar — CHANGES_REQUESTED** [review]
missing changelog for GA APIs. 
We should explicitly call out when we GA so many features.

**Review by kushagraThapar — APPROVED** [review]

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java (hunk 3 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosContainer.java:795]: We have it in beta now for over a year and recommended it to several customers - no way that we can make any breaking changes.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java (hunk 4 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/CosmosDiagnostics.java:115]: I don't want to remove anything in this release (walmart recommended - tight timeline) - let's remove deprecated APIs after holiday season - sometime in Q1

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroupConfigBuilder.java (hunk 5 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/ThroughputControlGroupConfigBuilder.java:14]: see above - let's deprecate after holiday season to not risk regression when shipping hotfixes

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMetaData.java (hunk 6 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMetaData.java:12]: Implementation package - no Beta needed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMode.java (hunk 7 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedMode.java:9]: Implementation package - no Beta needed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorItem.java (hunk 9 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/changefeed/common/ChangeFeedProcessorItem.java:17]: Implementation package - no Beta needed.

File: sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java (hunk 10 above)
Comment (FabianMeiswinkel) [reply sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosChangeFeedRequestOptions.java:88]: Nothing that could break these APIs - we have sipped it too long ago and are using for a while - so breaking it isn't an option IMO
